driver-core/src/main/com/mongodb/internal/connection/DefaultConnectionPool.java

```diff
@@ -168,18 +168,20 @@ public InternalConnection get(final OperationContext operationContext) {
 
     @Override
     public InternalConnection get(final OperationContext operationContext, final long timeoutValue, final TimeUnit timeUnit) {
-        connectionPoolListener.connectionCheckOutStarted(new ConnectionCheckOutStartedEvent(serverId));
+        connectionPoolListener.connectionCheckOutStarted(new ConnectionCheckOutStartedEvent(serverId, operationContext.getId()));
         Timeout timeout = Timeout.startNow(timeoutValue, timeUnit);
         try {
             stateAndGeneration.throwIfClosedOrPaused();
             PooledConnection connection = getPooledConnection(timeout);
             if (!connection.opened()) {
                 connection = openConcurrencyLimiter.openOrGetAvailable(connection, timeout);
             }
-            connectionPoolListener.connectionCheckedOut(new ConnectionCheckedOutEvent(getId(connection)));
+            connection.checkedOutForOperation(operationContext);
+            connectionPoolListener.connectionCheckedOut(new ConnectionCheckedOutEvent(getId(connection), operationContext.getId(),
+                    System.nanoTime() - timeout.getStartNanos()));
```

Comment:
`checkedOutForOperation` calls `System.nanoTime()` and uses it as the "start" for `ConnectionCheckedInEvent`, then we call `System.nanoTime()` and use it as the "stop" for `ConnectionCheckedOutEvent`. These "start" and "stop" values must always be equal, which is not currently the case.

There are many ways to achieve the above, the interesting part is to find a nice way of doing this while using the `Timer` abstraction proposed in [TODO @stIncMale].

Reply:
Removed elapsedTime from PR